lib/std/json.zig

```diff
@@ -132,6 +132,63 @@ pub const Token = union(enum) {
     Null,
 };
 
+// A FIFO bit-stack.
+fn Stack(comptime n: usize) type {
+    return struct {
+        const Self = @This();
+        const TypeInfo = std.builtin.TypeInfo;
+
+        const element_bitcount = 8 * @sizeOf(usize);
+        const element_count = n / element_bitcount;
+        const ElementType = @Type(TypeInfo{ .Int = TypeInfo.Int{ .signedness = .unsigned, .bits = element_bitcount } });
+
+        comptime {
+            std.debug.assert(n % element_bitcount == 0);
+        }
+
+        memory: [element_count]ElementType,
+        len: usize,
+
+        pub fn init(self: *Self) void {
+            self.memory = [_]ElementType{0} ** element_count;
+            self.len = 0;
+        }
+
+        pub fn push(self: *Self, bit: u1) !void {
```

Comment:
Reviewing myself, it might be a bit cleaner to return an optional for push/peek/pop and just use orelse when using the stack. We don't propagate any of these errors via try.